### PR TITLE
[Android] Implement Network Interceptors

### LINKED ISSUE #199: [Android] Implement Network Interceptors
**Parent Issue:** #7

**Description:**
Create interceptors for logging, error handling, and authentication.

**Tasks:**
- Create `LoggingInterceptor.java` for request/response logging
- Create `ErrorInterceptor.java` for error logging
- Create `AuthInterceptor.java` for device identification
- Add interceptors to OkHttpClient in NetworkModule
- Write unit tests

**Acceptance Criteria:**
- [ ] All interceptors created
- [ ] Properly configured in NetworkModule
- [ ] Error handling standardized
- [ ] 100% test coverage

### PR DESCRIPTION
**Pull request by PriyaBargota**
## Summary
Implements custom OkHttp interceptors for standardized network operations to replace the basic logging interceptor.

## Changes
- **LoggingInterceptor**: Detailed request/response logging with timing, header redaction, and one-shot body handling
- **ErrorInterceptor**: Standardized error logging for 4xx/5xx responses  
- **AuthInterceptor**: Device identification header injection
- Updated NetworkModule to use custom interceptors
- Comprehensive unit tests with 100% coverage

## Test Plan
- [x] All unit tests passing (41 test methods across 3 new test classes + 5 tests added to NetworkModuleTest)
- [x] Tests cover success cases, error cases, edge cases, and one-shot request bodies
- [x] Verified interceptor chain order (Auth → Logging → Error)

## Related Issues
Closes #199

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: android/app/src/main/java/com/manuscripta/student/network/interceptor/ErrorInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        return response;
+    }
+
+    /**
+     * Handles HTTP error responses by logging detailed error information.
+     *
+     * @param request  The HTTP request that resulted in an error
+     * @param response The error response
+     */
+    private void handleError(@NonNull Request request, @NonNull Response response) {
```
Comment (WillStephenn) [inline android/app/src/main/java/com/manuscripta/student/network/interceptor/ErrorInterceptor.java]: I feel like this function name is a bit misleading, it's not handling errors - it's logging them. It's calling `logClientError`
```suggestion
    private void logError(@NonNull Request request, @NonNull Response response) {
```
If this change is implemented, the call sites need updating too

File: android/app/src/main/java/com/manuscripta/student/network/interceptor/ErrorInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        String url = request.url().toString();
+
+        if (statusCode >= 400 && statusCode < 500) {
+            // Client errors (4xx)
+            logClientError(method, url, statusCode, response);
+        } else if (statusCode >= 500) {
+            // Server errors (5xx)
+            logServerError(method, url, statusCode, response);
+        }
+    }
```
Comment (WillStephenn) [inline android/app/src/main/java/com/manuscripta/student/network/interceptor/ErrorInterceptor.java]: Claude included `handleError` because the original issue said to handle errors. I think the errors should be handled in the repository layer, and the interceptors be responsible for logging only. What do you think @PriyaBargota? If you agree, we should rename this method to `logError`:

Comment (PriyaBargota) [reply android/app/src/main/java/com/manuscripta/student/network/interceptor/ErrorInterceptor.java]: I agree, the interceptor only logs, it doesn't handle errors. Makes sense to rename handleError() to logError() to match what it actually does. The repository layer is the right place for error handling. I will do this now.

**Review by WillStephenn — APPROVED** [review]
Merging this now @PriyaBargota
